### ISSUE TITLE
shellbtrfs.dll ReflinkCopyW - exception: access violation reading 0x0000000000000032

### ISSUE THREAD

**Issue by Fierelier** [maharmstone/btrfs#308]
I tried importing shellbtrfs.dll with python, and using its ReflinkCopyW function. Unfortunately, I am only getting 0x32. Using rundll32.exe, no errors are output (expected, no error handling) and no files are created.

Error:
```
Link: F:\Grand Theft Auto Episodes from Liberty City - umlMods\[1.1.2.0]\[3 Custom Mods]\timecyc++\pc\data\timecyc.dat -> F:\Grand Theft Auto Episodes from Liberty City - umlTemp\pc\data\timecyc.dat
Traceback (most recent call last):
  File "UniversalModloader.py", line 224, in <module>
    main()
  File "UniversalModloader.py", line 218, in main
    if choice == "Load mods": loadMods()
  File "UniversalModloader.py", line 150, in loadMods
    cloneFolder(ffile,pathTemp)
  File "UniversalModloader.py", line 133, in cloneFolder
    linkFile(ffile,nfile)
  File "UniversalModloader.py", line 107, in linkFile
    btrfsdll.ReflinkCopyW(src,dst)
OSError: exception: access violation reading 0x0000000000000032
Press ENTER to quit.
```

Code snippet:
```py
btrfsdll = False
def linkFile(src,dst,method = False):
	if not method: method = config["default"]["linkMethod"] 
	print("Link: " +src+ " -> " +dst)
	if method == "hardlink":
		os.link(src,dst)
		return
	
	if method == "copy":
		shutil.copy(src,dst)
		return
	
	if method == "reflink":
		if os.name == "nt":
			global btrfsdll
			if not btrfsdll: btrfsdll = ctypes.cdll.LoadLibrary("shellbtrfs.dll")
			btrfsdll.ReflinkCopyW(src,dst)
		else:
			subprocess.call(["cp","-n","--reflink=always",src,dst])
		return
	
	raise Exception("uml", "invalid link method")
```

Running Windows NT 6.1.7601 AMD64, Python 3.8.6 AMD64

**Comment by maharmstone** [comment]
I don't know Python, so I can't tell you what you _should_ be doing, but I can tell you that you're definitely calling ReflinkCopyW wrong, not least because it takes four parameters rather than two. Dig out the function prototype in src/shellext/contextmenu.cpp and make sure you're doing the same thing.

**Comment by Fierelier** [comment]
I had a look and I have tweaked my code to this:
```py
btrfsdll.ReflinkCopyW(0,0,'"' +src+ '" "' +dst+ '"',1)
```
I just passed 0 for the first 2 arguments, as I could not see them be used in the function (uncertain if that is valid), and I reshaped the src/dst to correspond to what I think is correct for the function. Now there is no error, but there is also no files, just like the rundll32 method.

If I use CMD and head into the game directory and enter `rundll32 shellbtrfs.dll,ReflinkCopy LaunchEFLC.exe LaunchEFLC2.exe` nothing happens, no file is created. I tried absolute paths as well, with quotes. I also tried the same as administrator. The shell extension works and I can paste as reflink.
